string file_name
string data
string width
string height
boolean immediately_apply
boolean overwrite_existing